packages/components/src/SvgIcon/SvgIcon.tsx

```diff
@@ -0,0 +1,106 @@
+// @flow
+import * as CSS from "csstype";
+import * as React from "react";
+import cx from "classnames";
+import styles from "./SvgIcon.module.css";
+
+type BaseProps = {
+  /**
+   * The SVG Color, expects a valid css color (hex, rgb, etc.). You can use
+   * Recommendation: if `currentColor` isn't sufficient, use color tokens from
+   * `import Tokens from '@chanzuckerberg/eds-tokens/json/variables.json';`
+   */
+  color?: string;
+  /**
+   * Either a fragment of svg elements (g, path, circle, etc.), or a single
+   * svg elment
+   */
+  children: React.ReactNode;
+  /**
+   * Width/Height string (px, rem, em, vh, etc.)
+   * Recommendation: use line-height tokens from
+   * `import Tokens from '@chanzuckerberg/eds-tokens/json/variables.json';`
+   */
+  size?: string;
+  /**
+   * Classname for additional custom styling in context.
+   */
+  className?: string;
+  /**
+   * The role of the icon. Should be 'presentation' for icons that don't provide
+   * any information on their own - for example, when an icon doesn't add any
+   * additional information to page text. Should be 'img' for icons that stand
+   * alone and have no other accompanying information - these will require you
+   * to provide `title` text that provides meaning to assistive technology.
+   */
+  role: "presentation" | "img";
+  /**
+   * Boolean, if true, the element will be displayed as a block, otherwise
+   * default is inline like images
+   */
+  block?: boolean;
+};
+
+type PresentationProps = BaseProps & {
+  role: "presentation";
+};
+
+type ImgProps = BaseProps & {
+  role: "img";
+  /**
+   * Accessible title that will be read out with screen readers.
+   */
+  title: string;
+};
+
+// Just for testing! The props being used should be SvgIconProps, but it's not passing
+// typescript right now.
+type TestProps = BaseProps & {
+  title?: string;
+};
```

Comment:
I think this is the last typescript error. If you use `SvgIconProps` below instead of `TestProps` and run `npx lerna run build:types`, you get this error:
```
src/SvgIcon/utils/createSvgIcon.tsx(22,8): error TS2322: Type '{ children: ReactNode; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' is not assignable to type '(IntrinsicAttributes & BaseProps & { role: "presentation"; } & RefAttributes<SVGElement>) | (IntrinsicAttributes & BaseProps & { ...; } & RefAttributes<...>)'.
  Property 'title' is missing in type '{ children: ReactNode; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' but required in type '{ role: "img"; title: string; }'.
```

We're trying to say that title is required if the role is "img" and not present if the role is "presentation" and typescript is not buying it.